pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -129,11 +130,37 @@ func (f *ring2Factory) PrintObject(cmd *cobra.Command, mapper meta.RESTMapper, o
 	return printer.PrintObj(obj, out)
 }
 
-func (f *ring2Factory) NewBuilder() *resource.Builder {
-	mapper, typer := f.objectMappingFactory.Object()
+func (f *ring2Factory) NewBuilder(local, useUnstructured bool) (*resource.Builder, meta.RESTMapper, error) {
+	var mapper meta.RESTMapper
+	var typer runtime.ObjectTyper
+	if useUnstructured {
+		var err error
+		mapper, typer, err = f.objectMappingFactory.UnstructuredObject()
+		// this may happen if apiserver cannot be contacted, but we need  --local commands to work
+		if err != nil {
+			if !local {
+				return nil, nil, err
+			}
+			mapper, typer = f.objectMappingFactory.Object()
```

Comment:
I'd rather figure out why we can't get an unstructured object than handle errors by ignoring the useUnstructured parameter… than seems like it would cause all sorts of problems